streamlit/main.py

```diff
@@ -14,81 +14,99 @@
+    if 'oldneu' not in st.session_state:
+        # contagem de neutros
+        st.session_state.oldneu = 0
     if 'negative' not in st.session_state:
+        # contagem de negativos
         st.session_state.negative = 0
-    if 'trigger' not in st.session_state:
-        st.session_state.trigger = 0
-
-
+    if 'oldneg' not in st.session_state:
+        # contagem de negativos
+        st.session_state.oldneg = 0
+    if 'df' not in st.session_state:
+        # dataframe para o display e download dos resultados
+        st.session_state.df = pd.DataFrame(st.session_state.feedList)
 
 
+    #importando as 2 apis para o fuzzy comparar
     goodList = apiGood()
     badList = apiBad()
 
-    st.image("https://aedv.es/wp-content/uploads/2020/06/encuesta-aedv-1024x512.jpg")
-
+    # titulo e foto do site
+    st.image("https://aedv.es/wp-content/uploads/2020/06/encuesta-aedv-1024x512.jpg",width=395)
     st.title("Feedback Calculator")
-    st.cache(func=None, persist=False, allow_output_mutation=False, show_spinner=True, suppress_st_warning=False,
-             hash_funcs=None, max_entries=None, ttl=None)
 
-    tab1, tab2 = st.tabs(["Import Data", "Results"])
+    # divisao de 3 paginas, sendo para importar os dados, ver os resultados e um tutorial de como usar o site
+    tab1, tab2 , tab3= st.tabs(["Import Data", "Results","Tutorial"])
     with tab1:
         st.header("Import Data")
-        # upload text
+        # feature para usuario fazer upload manual(escrever o feedback)
         text = st.text_area("Single Feedback")
-        # upload csv
-        data = st.file_uploader("Multiple Feedback", type="csv")
-
-
-        if st.button("Add"):
-            st.session_state.trigger = 1
 
-            if data is not None:
+        # feature para usuario fazer upload em csv
+        csv = st.file_uploader("Multiple Feedback", type="csv")
 
-                df = pd.read_csv(data, sep=';')
-                list = df.values.tolist()
+        # true se o usuario fizer o upload de um csv
+        if csv is not None:
+            # e apertar o botao
+            if st.button("Add"):
+                # le o csv e armazena em um df, depois converte para uma lista
+                idf = pd.read_csv(csv, sep=';')
+                list = idf.values.tolist()
 
+                # para cada i no tamanho da lista, da um append na lista de feedbacks totais
                 for i in range(len(list)):
                     st.session_state.feedList.append(unidecode(list[i][0]))
-                data = None
-            else:
-                st.session_state.feedList.append(unidecode(text))
-                text = None
 
+                # apos o procedimento esvazia o input para nao ter dados duplicados
+                csv = None
 
-            if st.session_state.trigger == 1:
                 st.success("Feedback Added!")
-                time.sleep(5)
-                st.session_state.trigger = 0
+                time.sleep(3)
```

Comment:
🤔, porque precisamos ter o `sleep` aqui ??

Reply:
quando o dado for inserido mensagem positiva "Feedback Added!" aparece.
e se nada acontecer, a mensagem fica la pra sempre
tanto que essa variável trigger era eu brigando pra descobrir uma forma de tirar a mensagem depois de um tempo

a solução foi esperar uns segundos e usar a função "st.experimental_rerun()"

Porque a logica do streamlit, diz que quando um botao for pressionado, ele apenas se torna falso quando o script da um rerun, e isso ocorre quando outra funcao é iniciada, tipo apertar um botao, inserir um dado etc.

a ideia era forçar isso se passando alguns segundos dps da ação ser concluida

Reply:
ah, saquei 🚀